Source/WebCore/history/CachedFrame.cpp

```diff
@@ -266,7 +283,8 @@ void CachedFrame::destroy()
     if (m_cachedFramePlatformData)
         m_cachedFramePlatformData->clear();
 
-    LocalFrame::clearTimers(m_view.get(), m_document.get());
+    if (RefPtr localFrameView = dynamicDowncast<LocalFrameView>(m_view.get()))
+        LocalFrame::clearTimers(localFrameView.get(), m_document.get());
```

Comment:
Here I think you might dereference document in LocalFrame::clearTimers if the localframe doesn't have one.